Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -278,6 +278,50 @@ void ReflectometryReductionOne::init() {
                   Direction::Input);
 }
 
+//----------------------------------------------------------------------------------------------
+/** Validate the input properties.
+*/
+std::map<std::string, std::string> ReflectometryReductionOne::validateInputs() {
+
+  std::map<std::string, std::string> result;
+
+  MatrixWorkspace_sptr decWS = getProperty("DetectorEfficiencyCorrection");
+  MatrixWorkspace_sptr runWS = getProperty("InputWorkspace");
+
+  if (!decWS) {
+    return result;
+  }
+
+  if (decWS->getAxis(0)->unit()->unitID() != "Wavelength") {
+    result["DetectorEfficiencyCorrection"] = "Detector Efficiency Correction "
+      "workspace x-units must be in wavelength";
+  }
+  else if (decWS->getNumberHistograms() > 1 &&
+      decWS->getNumberHistograms() < runWS->getNumberHistograms()) {
+    result["DetectorEfficiencyCorrection"] = "The number of spectra in the "
+      "Detector Efficiency Correction workspace must either be 1 or equal to "
+      "the number of spectra in the input workspace";
+  } else {
+    double firstVal = decWS->y(0)[0];
+    if (firstVal == 0) {
+      result["DetectorEfficiencyCorrection"] = "Detector Efficiency Correction "
+        "workspace cannot contain any zero values.";
+    } else {
+      for (size_t i = 1; i < decWS->getNumberHistograms(); i++) {
+        if (decWS->y(i)[0] != firstVal) {
+          result["DetectorEfficiencyCorrection"] = "All values in the Detector "
+            "Efficiciency Correction Workspace must be equal";
+          break;
```

Comment:
Values in the `DetectorEfficiencyCorrection` workspace don't need to be the same. In fact, in most cases they will be different, as the idea is that this workspace will be used to normalize each spectrum by a different normalization constant, taking into account possible differences in efficiencies. Besides, if all numbers were the same, why would we need to have more than one spectrum? One would be enough if that was the case. The above `else` is not what we want. 

Aso, we don't need to check that values are non-zero. If a user inputs a detector efficiency workspace full of zeroes, they will get `NaN`s in the output workspace. That should be enough to warn them that they are inputing something incorrect. The `if (firstVal == 0)` can be removed as well from the checks.